src/main/java/hudson/plugins/ec2/AmazonEC2Cloud.java

```diff
@@ -185,14 +185,20 @@ public FormValidation doCheckAltEC2Endpoint(@QueryParameter String value) {
         public ListBoxModel doFillRegionItems(
                 @QueryParameter String altEC2Endpoint,
                 @QueryParameter boolean useInstanceProfileForCredentials,
+                @QueryParameter String roleArn,
+                @QueryParameter String roleSessionName,
+                @QueryParameter String region,
                 @QueryParameter String credentialsId)
 
                 throws IOException, ServletException {
             ListBoxModel model = new ListBoxModel();
             if (Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
                 try {
                     AWSCredentialsProvider credentialsProvider = createCredentialsProvider(useInstanceProfileForCredentials,
```

Comment:
as far as I can tell from EC2Cloud.java my use here should be OK because there is a guard for empty roleArn and roleSessionName

https://github.com/jenkinsci/ec2-plugin/blob/master/src/main/java/hudson/plugins/ec2/EC2Cloud.java#L956

```
        if (StringUtils.isNotEmpty(roleArn) && StringUtils.isNotEmpty(roleSessionName)) {
            return new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
...
```